app/scripts/modules/core/src/pipeline/config/triggers/artifacts/gcs/gcs.artifact.ts

```diff
@@ -0,0 +1,45 @@
+import { IController, module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
+import { IArtifact } from 'core/domain/IArtifact';
+import { PipelineConfigProvider } from 'core/pipeline';
+
+class GcsArtifactController implements IController {
+  constructor(public artifact: IArtifact) {
+    'ngInject';
+  }
+
+  public onInit(): void {
```

Comment:
I don't love that this is an implicitly required method for an artifact controller. I would either just do this kind of initialization inside the constructor or have these controllers implement an interface (though you would have to not do the `any` casting in `loadArtifactKind` above).

Reply:
Let me double check that the controller does get constructed each time, otherwise you can get in a bad state where selecting some special artifact leaves various important fields filled out incorrectly.